### ISSUE TITLE
goto $coderef should be documented

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13035]
Migrated from [rt.perl.org#118523](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=118523) (status was 'resolved')

Searchable as RT118523$

**Comment by p5pRT** [comment]
### From @mauke

#### Created by @mauke



perldoc \-f goto claims that goto has three forms&#8203;:
\- goto LABEL  \(jumps to a label\)
\- goto EXPR   \(jumps to a dynamically computed label\)
\- goto &NAME  \(performs a tail call\)

Experiments suggest that 'goto EXPR' performs a tail call \(i\.e\. is equivalent
to 'goto &\{ EXPR \}'\) if EXPR evaluates to a code reference\. This should be
documented under 'goto EXPR'\.

Also\, 'goto \_\_SUB\_\_' deserves explicit mentioning \(it's parsed as an EXPR\, not
a LABEL\, and because it evaluates to a coderef\, it behaves like 'goto
&\{\+\_\_SUB\_\_\}'\)\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=low

This perlbug was built using Perl 5.12.1 - Thu Jun  3 20:09:15 CEST 2010
It is being executed now by  Perl 5.18.0 - Fri May 24 23:20:48 CEST 2013.

Site configuration information for perl 5.18.0:

Configured by mauke at Fri May 24 23:20:48 CEST 2013.

Summary of my perl5 (revision 5 version 18 subversion 0) configuration:
   
  Platform:
    osname=linux, osvers=3.5.7-gentoo, archname=i686-linux
    uname='linux nora 3.5.7-gentoo #5 preempt sat jan 26 16:46:10 cet 2013 i686 amd athlon(tm) 64 processor 3200+ authenticamd gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -flto',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.8.0', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-fstack-protector -L/usr/local/lib -O2 -flto'
    libpth=/usr/local/lib /lib/../lib /usr/lib/../lib /lib /usr/lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.15.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.15'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -flto -L/usr/local/lib -fstack-protector'

Locally applied patches:
    SAVEARGV0 - disable magic open in <ARGV>


@INC for perl 5.18.0:
    /home/mauke/usr/local/lib/perl5/site_perl/5.18.0/i686-linux
    /home/mauke/usr/local/lib/perl5/site_perl/5.18.0
    /home/mauke/usr/local/lib/perl5/5.18.0/i686-linux
    /home/mauke/usr/local/lib/perl5/5.18.0
    .


Environment for perl 5.18.0:
    HOME=/home/mauke
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=POSIX
    LD_LIBRARY_PATH=/home/mauke/usr/local/lib
    LOGDIR (unset)
    PATH=/home/mauke/usr/perlbrew/bin:/home/mauke/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/opt/bin:/usr/i686-pc-linux-gnu/gcc-bin/4.6.3:/opt/sun-jdk-1.4.2.13/bin:/opt/sun-jdk-1.4.2.13/jre/bin:/opt/sun-jdk-1.4.2.13/jre/javaws:/opt/dmd/bin:/usr/games/bin
    PERLBREW_BASHRC_VERSION=0.43
    PERLBREW_HOME=/home/mauke/.perlbrew
    PERLBREW_PATH=/home/mauke/usr/perlbrew/bin
    PERLBREW_ROOT=/home/mauke/usr/perlbrew
    PERLBREW_VERSION=0.27
    PERL_BADLANG (unset)
    PERL_UNICODE=SAL
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @rjbs

I propose to apply 
http&#8203;://perl5\.git\.perl\.org/perl\.git/blobdiff/21339399ecf6ae3a636f01859c6fd863bfac6933\.\.4e4f
9631605e6dc9337e60f316d7c00df9a22feb&#8203;:/pod/perlfunc\.pod to blead if no one objects\.
 
\-\- 
rjbs

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rjbs

Resolved by 3e8a637\.
 
\-\- 
rjbs

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Resolved by 3e8a637\.
 
\-\- 
rjbs

**Comment by p5pRT** [comment]
@rjbs - Status changed from 'open' to 'resolved'
